File: backend/treinamentoPython/cursoPython/jogos/adivinhacao.py
def jogar_adivinhacao():
    import random

    print("################################")
    print("Bem vindo no jogo de adivinhacao")
    print("################################\n")

    numero_secreto = random.randrange(1, 101)
    numero_maior = 100
    numero_menor = 1
    total_tentativas = 0
    pontos = 1000

    print("Qual o nível de dificuldade?")
    print("(1) Fácil (2) Médio (3) Difícl")

    nivel = int(input("Digite o nível: "))

    while (nivel <= 0 or nivel >= 3):
        print("\nNível invalido!")
        print("(1) Fácil (2) Médio (3) Difícl")
        nivel = int(input("\nDigite o nível: "))
        
    if (nivel == 1):
        total_tentativas = 20
    elif (nivel == 2):
        total_tentativas = 10
    elif (nivel == 3):
        total_tentativas = 5
        

    for rodada in range(1,total_tentativas + 1):
        print(f"\nTentativa: {rodada} de {total_tentativas}" )
        chute_str  = input("Digite seu número entre 1 e 100: ")

        chute = int(chute_str)

        if (chute < 1 or chute > 100):
            print("Você deve digitar um número entre 1 e 100")
            continue
            
        acertou  = chute ==  numero_secreto
        maior = chute > numero_secreto
        menor  = chute < numero_secreto
        
        if (acertou):
            print(f"\nVocê acertou! E seu score foi {pontos}")
            break
        else:
            if (maior):
                numero_maior = chute
                print("\nVocê errou! Seu chute foi maior que o do número secreto!")
                print(f"Seu número está entre {numero_menor} e {numero_maior}")
            elif (menor):
                numero_menor = chute
                print("\nVocê errou! Seu chute foi menor que o do número secreto!")
                print(f"Seu número está entre {numero_menor} e {numero_maior}")
            pontos_perdidos = abs(numero_menor - chute)
            pontos -= pontos_perdidos
        

    print("\nFim do jogo!")

if(__name__ == "__main__"):
    jogar_adivinhacao()

File: desktop/primeiro_curso_de_C/adivinhacao/adivinhacao.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main(void) {

    printf("\n\n");
    printf("         P  /_\\  P                             \n");                     
    printf("        /_\\_|_|_/_\\                           \n");
    printf("    n_n | ||. .|| | n_n         Bem vindo ao    \n");
    printf("    |_|_|nnnn nnnn|_|_|     Jogo de Adivinhação!\n");
    printf("   |" "  |  |_|  |"  " |                        \n");
    printf("   |_____| ' _ ' |_____|                        \n");
    printf("         \\__|_|__/                             \n\n\n");

    int segundos = time(0);

    srand(segundos);

    int numero_secreto = rand() % 100;
    int tentativas = 1;

    int acertou = 0;

    int nivel;
    printf("Qual o nível de dificuldade? \n");
    printf("(1) Fácil (2) Médio (3) Dificil\n\n");
    printf("Escolha: ");
    scanf("%d", &nivel);

    int numero_tentativas;
    
    switch(nivel) {
        case 1:
            numero_tentativas = 20;
            break;
        case 2:
            numero_tentativas = 15;
            break;
        default:
            numero_tentativas = 6;
            break;
    }

    double pontos = 1000; 

    // printf("\nO número %d é o secreto. Não conta para ninguém!\n", numero_secreto);

    // Pedindo um chute ao usuário
    for (int i = 1; i <= numero_tentativas; i++) {
        int chute;

        printf("Tentativa %d\n", i);

        printf("Qual é o seu chute? ");
        scanf("%d", &chute);
        printf("Seu chute foi %d\n", chute);

        if (chute < 0) {
            printf("Você não pode chuter números negativos!");

            continue;
        }

        acertou = (numero_secreto == chute);
        int maior = chute > numero_secreto;

        if (acertou) {
            break;
        }

        else if (maior) {
            printf("Seu chute foi maior que o número secreto\n");
        }

        else {
            printf("Seu chute foi menor que o número secreto\n");
        }

        tentativas++;

        double pontosperdidos = (abs)(chute - numero_secreto) / (double)2;
        pontos  = pontos - pontosperdidos;
    }

    printf("Fim de jogo!\n");

    if (acertou) {

        printf("\n\n\n");
        printf("                   OOOOOOOOOOO                   \n");
        printf("                OOOOOOOOOOOOOOOOOOO              \n"); 
        printf("              OOOOOO  OOOOOOOOO  OOOOOO          \n");
        printf("           OOOOOO      OOOOO      OOOOOO         \n");
        printf("         OOOOOOOO  #   OOOOO  #   OOOOOOOO       \n");
        printf("        OOOOOOOOOO    OOOOOOO    OOOOOOOOOO      \n");
        printf("        OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO    \n");
        printf("        OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO    \n");
        printf("        OOOO  OOOOOOOOOOOOOOOOOOOOOOOOO  OOOO    \n");
        printf("         OOOO  OOOOOOOOOOOOOOOOOOOOOOO  OOOO     \n");
        printf("         OOOO   OOOOOOOOOOOOOOOOOOOO  OOOO       \n");
        printf("           OOOOO   OOOOOOOOOOOOOOO   OOOO        \n");
        printf("            OOOOOO   OOOOOOOOO   OOOOOO          \n");
        printf("              OOOOOO         OOOOOO              \n");
        printf("                    OOOOOOOOOOOO                 \n\n\n");

        printf("Você ganhou!\n");
        printf("Você acertou em %d tentativas!\n", tentativas);
        printf("Total de pontos %.1f\n", pontos);  
    } else {        
        printf("Você perdeu tente novamente!\n");

        printf("       \\|/ ____ \\|/    \n");
        printf("        @~/ ,. \\~@      \n");
        printf("       /_( \\__/ )_\\    \n");
        printf("          \\__U_/        \n");
    }

   
}   